clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -1124,17 +1125,37 @@ public void close() {
      * @param timeUnit The time unit for the <code>timeout</code>
      * @throws InterruptException If the thread is interrupted while blocked
      * @throws IllegalArgumentException If the <code>timeout</code> is negative.
+     *
+     * @deprecated Since 2.1. Use {@link #close(Duration)} or {@link #close()}.
      */
+    @Deprecated
     @Override
     public void close(long timeout, TimeUnit timeUnit) {
-        close(timeout, timeUnit, false);
+        close(Duration.ofMillis(timeUnit.toMillis(timeout)));
     }
 
-    private void close(long timeout, TimeUnit timeUnit, boolean swallowException) {
-        if (timeout < 0)
-            throw new IllegalArgumentException("The timeout cannot be negative.");
+    /**
+     * This method waits up to <code>timeout</code> for the producer to complete the sending of all incomplete requests.
+     * <p>
+     * If the producer is unable to complete all requests before the timeout expires, this method will fail
+     * any unsent and unacknowledged records immediately.
+     * <p>
+     * If invoked from within a {@link Callback} this method will not block and will be equivalent to
+     * <code>close(Duration.ofMillis(0))</code>. This is done since no further sending will happen while
+     * blocking the I/O thread of the producer.
+     *
+     * @param timeout The maximum time to wait for producer to complete any pending requests. The value should be
+     *                non-negative. Specifying a timeout of zero means do not wait for pending send requests to complete.
+     * @throws InterruptException If the thread is interrupted while blocked
+     * @throws IllegalArgumentException If the <code>timeout</code> is negative.
+     */
+    @Override
+    public void close(Duration timeout) {
+        close(timeout, false);
+    }
 
-        long timeoutMs = timeUnit.toMillis(timeout);
+    private void close(Duration timeout, boolean swallowException) {
+        long timeoutMs = timeout.toMillis();
```

Comment:
Do we need to add negative timeout check like we did in [consumer](https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java#L2101)?